tests/e2e/create/create_test.go

```diff
@@ -0,0 +1,456 @@
+				registry:         ociRegistry,
+				insecure:         true,
+				output:           templateOutputPath,
+			}
+		})
+		It("Then the command should succeed", func() {
+			Expect(cmd.execute()).To(Succeed())
+
+			By("And module template file should be generated")
+			Expect(filesIn("/tmp/")).Should(ContainElement("template.yaml"))
+		})
+		It("Then module template should contain the expected content", func() {
+			template, err := readModuleTemplate(templateOutputPath)
+			Expect(err).ToNot(HaveOccurred())
+			descriptor := getDescriptor(template)
+			Expect(descriptor).ToNot(BeNil())
+			Expect(descriptor.SchemaVersion()).To(Equal(compdescv2.SchemaVersion))
+
+			By("And annotations should be correct")
+			annotations := template.Annotations
+			Expect(annotations[shared.ModuleVersionAnnotation]).To(Equal("1.0.0"))
+			Expect(annotations[shared.IsClusterScopedAnnotation]).To(Equal("false"))
+
+			By("And descriptor.component.repositoryContexts should be correct")
+			Expect(descriptor.RepositoryContexts).To(HaveLen(1))
+			repo := descriptor.GetEffectiveRepositoryContext()
+			Expect(repo.Object["baseUrl"]).To(Equal(ociRegistry))
+			Expect(repo.Object["componentNameMapping"]).To(Equal(string(ocmocireg.OCIRegistryURLPathMapping)))
+			Expect(repo.Object["type"]).To(Equal(ocireg.Type))
+
+			By("And descriptor.component.resources should be correct")
+			Expect(descriptor.Resources).To(HaveLen(1))
+			resource := descriptor.Resources[0]
+			Expect(resource.Name).To(Equal("raw-manifest"))
+			Expect(resource.Relation).To(Equal(ocmmetav1.LocalRelation))
+			Expect(resource.Type).To(Equal("directory"))
+			Expect(resource.Version).To(Equal("1.0.0"))
+
+			By("And descriptor.component.resources[0].access should be correct")
+			resourceAccessSpec1, err := ocm.DefaultContext().AccessSpecForSpec(descriptor.Resources[0].Access)
+			Expect(err).ToNot(HaveOccurred())
+			localBlobAccessSpec, ok := resourceAccessSpec1.(*localblob.AccessSpec)
+			Expect(ok).To(BeTrue())
+			Expect(localBlobAccessSpec.GetType()).To(Equal(localblob.Type))
+			Expect(localBlobAccessSpec.LocalReference).To(ContainSubstring("sha256:"))
+			Expect(localBlobAccessSpec.MediaType).To(Equal("application/x-tar"))
+
+			By("And descriptor.component.sources should be empty")
+			Expect(len(descriptor.Sources)).To(Equal(0))
+
+			By("And spec.mandatory should be false")
+			Expect(template.Spec.Mandatory).To(BeFalse())
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with same version that already exists in the registry", func() {
+			cmd = createCmd{
+				moduleConfigFile: minimalConfig,
+				registry:         ociRegistry,
+				insecure:         true,
+			}
+		})
+		It("Then the command should fail with same version exists message", func() {
+			err := cmd.execute()
+			Expect(err.Error()).Should(ContainSubstring("could not push component version: component version already exists"))
+		})
+	})
```

Comment:
Is my understanding correct that we are not supporting `--module-archive-version-overwrite` anymore?

Reply:
Yes, the flag has been removed. You can only push a version once now. Any subsequent push with the same version should fail.